hasher-matcher-actioner/hmalib/lambdas/api/api_root.py

```diff
@@ -135,6 +133,22 @@ def submit_content_request_from_s3_event_record(
     )
 
 
+def get_content_id_from_bucket_and_key(bucket: str, key: str) -> str:
+    """
+    For partner buckets, we use the full bucket name and key as the content ID to avoid
+    ContentID collisions between buckets.
+
+    There is still a threat of collision between partner bucket submissions and API/UI
+    submissions which is why we encourage (but dont enforce) partners to only use a single
+    submission method in production
+
+    The Matches UI page will work incorrectly if the content id includes / or ? so we replace
+    These chars with . when generating content ids.
+    """
+
+    return bucket + ":" + key.replace("/", ".").replace("?", ".")
```

Comment:
When actions are called for s3 objects matched, we will have to reverse this.

How about fixing the matches UI page instead? There is nothing inherently unsupportable about content_ids containing special characters.

Reply:
Also, reversing might become impossible if you are replacing with "." because file names will possibly contain ".".

The use case I am thinking of:
- partner processes a bucket.
- receives some matches on their webhook. webhook mentions content_id as partner_bucket:directory1.directory2.filename.filename.png
- how do they map back to the actual s3 key if we are 'destroying' the s3 key and not storing a copy